packages/sit-onyx/src/components/OnyxRadioButton/OnyxRadioButton.vue

```diff
@@ -8,34 +9,124 @@ export type RadioButtonProps<TValue> = SelectionProps<TValue> & {
    * See also: https://html.spec.whatwg.org/multipage/input.html#radio-button-group
    */
   name: string;
+  isRequired?: boolean;
+  errorMessage?: string;
 };
 
 const props = defineProps<RadioButtonProps<TValue>>();
+
+const selectorRef = ref<HTMLInputElement>();
+
+watchEffect(() => selectorRef.value?.setCustomValidity(props.errorMessage ?? ""));
 </script>
 
 <template>
   <!-- TODO: decide on support prefix and/or folder -->
-  <label class="onyx-radio-button">
+  <label class="onyx-radio-button" :title="props.errorMessage">
     <!-- TODO: readonly is not supported on native radio input: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-readonly#description -->
+    <!-- TODO: accessible error: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage -->
     <input
+      ref="selectorRef"
       class="onyx-radio-button__selector"
       type="radio"
+      :required="props.isRequired"
       :name="props.name"
       :value="props.id"
       :checked="props.selected"
-      :disabled="props.isDisabled"
+      :disabled="props.isDisabled || props.isReadonly"
     />
     <span class="onyx-radio-button__label">{{ props.label }}</span>
   </label>
 </template>
 
 <style lang="scss">
 .onyx-radio-button {
```

Comment:
I'd prefer to define the custom CSS variables and their fallback/default value at the top. Imho its better understandable whats going on. Example:

Currently:
```css
  &:has(&__selector:hover) {
    --onyx-radio-button-selector-border-color: var(--onyx-color-base-primary-300);
  }

    border: {
      style: solid;
      width: 1px;
      color: var(--onyx-radio-button-selector-border-color, var(--onyx-color-base-neutral-400));
    }
```


Suggestion:
```css
--onyx-radio-button-selector-border-color: var(--onyx-color-base-neutral-400);
// other custom CSS variable here

  &:has(&__selector:hover) {
    --onyx-radio-button-selector-border-color: var(--onyx-color-base-primary-300);
  }

    border: {
      style: solid;
      width: 1px;
      color: var(--onyx-radio-button-selector-border-color);
    }
```

Reply:
done, not sure if I like it tough 🤔  Lets keep it for now :)